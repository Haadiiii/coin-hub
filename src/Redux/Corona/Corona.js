import { createAsyncThunk } from '@reduxjs/toolkit';

const CORONA_SUCCESS = 'CORONA_SUCCESS';
const CORONA_BY_COUNTRY_SUCCESS = 'CORONA_BY_COUNTRY_SUCCESS';

const url = 'https://covid-api.mmediagroup.fr/v1/cases/';
const initialState = {
  coronaArray: [],
  coronaById: {},
};

export const getCorona = createAsyncThunk(
  CORONA_SUCCESS,
  async (args, { dispatch }) => {
    const response = await fetch(url);
    const data = await response.json();
    const coronas = Object.keys(data)
      .map((key) => {
        const virus = data[key].All;
        return {
          country: key,
          ...virus,
        };
      });
    dispatch({ type: CORONA_SUCCESS, payload: coronas });
    return coronas;
  },
);

const urlMain = 'https://covid-api.mmediagroup.fr/v1/cases?country=';

export const getCoronaByName = (country) => async (dispatch) => {
  const response = await fetch(
    `${urlMain}${country}`,
  );
  const data = await response.json();
  dispatch({ type: CORONA_BY_COUNTRY_SUCCESS, payload: data.All });
};

const coronaReducer = (state = initialState, action) => {
  switch (action.type) {
    case CORONA_SUCCESS:
      return {
        ...state,
        coronaArray: action.payload,
      };
    case CORONA_BY_COUNTRY_SUCCESS:
      return {
        ...state,
        coronaById: action.payload,
      };
    default:
      return state;
  }
};

export default coronaReducer;
