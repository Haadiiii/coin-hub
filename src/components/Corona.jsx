import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import uuid from 'react-uuid';
import { BiSearch } from 'react-icons/bi';
import { getCorona } from '../Redux/Corona/Corona';
import '../Styles/corona.css';

const Corona = () => {
  const { coronaArray } = useSelector((state) => state.corona);
  const [search, setSearch] = useState('');
  const [select, setSelect] = useState('');
  const navigate = useNavigate();
  const dispatch = useDispatch();
  useEffect(() => {
    if (coronaArray.length === 0) {
      dispatch(getCorona());
    }
  }, []);

  const handleSearch = (e) => {
    setSearch(e.target.value);
  };

  const handleSelect = (e) => {
    setSelect(e.target.value);
  };
  return (
    <div>
      <div className="search-Bar">
        <div className="field">
          <input
            type="text"
            value={search}
            placeholder="Search"
            onChange={handleSearch}
          />
          {' '}
          <BiSearch className="icon" />
        </div>
        <select value={select} onChange={handleSelect}>
          <option value="">Country</option>
          {coronaArray.map((item) => (
            <option key={uuid()} value={item.country}>
              {item.country}
            </option>
          ))}
        </select>
      </div>
      <div className="imagess">
        <img
          src="https://www.afge.org/link/c5634a73b41b4dd8a3f48e6768a6422d.aspx"
          alt=""
        />
      </div>
      <div className="stats">
        <p>Stats By country </p>
      </div>
      <div className="container">
        {coronaArray
          .filter((corona) => {
            if (search === '' && select === '') {
              return corona;
            } if (
              corona.country.toLowerCase().includes(search.toLowerCase())
              && corona.country.toLowerCase().includes(select.toLowerCase())
            ) {
              return corona;
            }
            return null;
          })
          .map((item) => (
            <div
              key={uuid()}
              style={{ width: '18rem' }}
              className="mb-2"
              onClick={() => navigate(`/Details/${item.country}`)}
            >
              <h2>&#10148;</h2>
              <div className="body">
                <div className="card-t">
                  <h3>
                    {item.country}
                    {' '}
                  </h3>
                  <p>
                    Cases:
                    {item.confirmed}
                  </p>
                </div>
              </div>
            </div>
          ))}
      </div>
    </div>
  );
};

export default Corona;
