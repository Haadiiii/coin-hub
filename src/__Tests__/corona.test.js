import React from 'react';
import renderer from 'react-test-renderer';
import Corona from './__mock__/corona';

describe('Details', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Corona />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
