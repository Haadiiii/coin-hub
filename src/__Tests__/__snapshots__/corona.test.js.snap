// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders correctly 1`] = `
<div>
  <div
    className="imagess"
  >
    <img
      alt=""
      src="https://www.afge.org/link/c5634a73b41b4dd8a3f48e6768a6422d.aspx"
    />
  </div>
  <div
    className="container"
  >
    <div
      className="mb-2"
      onClick={[Function]}
      style={
        {
          "width": "18rem",
        }
      }
    >
      <h2>
        âž¤
      </h2>
      <div
        className="body"
      >
        <div
          className="card-t"
        >
          <h3>
             
          </h3>
          <p>
            Population:
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
